Nio/Conversations/MessageComposerView.swift

```diff
@@ -1,40 +1,88 @@
 import SwiftUI
 
+struct ExDivider: View {
+    let color: Color = .accentColor
+    let width: CGFloat = 3
+    var body: some View {
+        Rectangle()
+            .fill(color)
+            .frame(width: width)
+            .edgesIgnoringSafeArea(.horizontal)
+    }
+}
+
 struct MessageComposerView: View {
     @Environment (\.colorScheme) var colorScheme
+    @Environment(\.colorSchemeContrast) var colorSchemeContrast
+    @Environment(\.sizeCategory) var sizeCategory
 
     @Binding var message: String
     @Binding var showAttachmentPicker: Bool
 
     var onCommit: () -> Void
 
-    var body: some View {
-        HStack {
-            Button(action: {
-                self.showAttachmentPicker.toggle()
-            }, label: {
-                Image(systemName: "paperclip")
-                    .font(.system(size: 20))
-                    .accessibility(label: Text(L10n.Composer.AccessibilityLabel.sendFile))
-            })
+    var highlightMessage: String?
+    var onCancel: () -> Void
 
-            ZStack {
-                Capsule(style: .continuous)
-                    .frame(height: 40)
-                    .foregroundColor(colorScheme == .light ? Color(#colorLiteral(red: 0.9332506061, green: 0.937307477, blue: 0.9410644174, alpha: 1)) : Color(#colorLiteral(red: 0.2549019754, green: 0.2745098174, blue: 0.3019607961, alpha: 1)))
+    var textColor: Color {
+        return .primaryText(for: colorScheme, with: colorSchemeContrast)
+    }
 
-                TextField(L10n.Composer.newMessage, text: $message, onCommit: onCommit)
-                    .padding(.horizontal)
+    var body: some View {
+        VStack {
+            if self.highlightMessage != nil {
+                Divider()
+                HStack {
+                    ExDivider()
+                        .background(Color.accentColor)
+                    VStack {
+                        HStack {
+                            Text(L10n.Composer.editMessage)
+                                .frame(minWidth: 0, maxWidth: .infinity, alignment: Alignment.leading)
+                                .padding(.leading, 10)
+                                .foregroundColor(.accentColor)
+                            Button(action: {
+                                self.onCancel()
+                            }, label: {
+                                Image(systemName: "multiply")
+                                    .font(.system(size: 20))
+                                    .accessibility(label: Text(L10n.Composer.AccessibilityLabel.cancelEdit))
+                            })
+                        }
+                        Text(highlightMessage!)
+                            .lineLimit(2)
+                            .padding(Edge.Set([.leading, .trailing, .bottom]), 10)
```

Comment:
This can be simplified a bit.
```suggestion
                            .padding([.horizontal, .bottom], 10)
```